patches/server/0910-Damage-with-any-cause.patch

```diff
@@ -0,0 +1,483 @@
+             CraftEventFactory.blockDamage = null; // SPIGOT-6639: Clear blockDamage to allow other entity damage during event call
+             CraftEventFactory.callEvent(event);
+-            CraftEventFactory.blockDamage = damager; // SPIGOT-6639: Re-set blockDamage so that other entities which are also getting damaged have the right cause
++            CraftEventFactory.blockDamage = blockDamager; // SPIGOT-6639: Re-set blockDamage so that other entities which are also getting damaged have the right cause // Paper
+ 
+             if (!event.isCancelled()) {
+                 event.getEntity().setLastDamageCause(event);
+@@ -1035,7 +1043,7 @@ public class CraftEventFactory {
+             return event;
+         } else if (CraftEventFactory.blockDamage != null) {
+             DamageCause cause = null;
+-            Block damager = CraftEventFactory.blockDamage;
++            Block blockDamager = CraftEventFactory.blockDamage; // Paper
+             if (source == DamageSource.CACTUS || source == DamageSource.SWEET_BERRY_BUSH || source == DamageSource.STALAGMITE || source == DamageSource.FALLING_STALACTITE || source == DamageSource.ANVIL) {
+                 cause = DamageCause.CONTACT;
+             } else if (source == DamageSource.HOT_FLOOR) {
+@@ -1047,12 +1055,13 @@ public class CraftEventFactory {
+             } else {
+                 throw new IllegalStateException(String.format("Unhandled damage of %s by %s from %s", entity, damager, source.msgId));
+             }
+-            EntityDamageEvent event = new EntityDamageByBlockEvent(damager, entity.getBukkitEntity(), cause, modifiers, modifierFunctions);
++            /*
++            EntityDamageEvent event = new EntityDamageByBlockEvent(blockDamager, entity.getBukkitEntity(), cause, modifiers, modifierFunctions); // Paper
+             event.setCancelled(cancelled);
+ 
+             CraftEventFactory.blockDamage = null; // SPIGOT-6639: Clear blockDamage to allow other entity damage during event call
+             CraftEventFactory.callEvent(event);
+-            CraftEventFactory.blockDamage = damager; // SPIGOT-6639: Re-set blockDamage so that other entities which are also getting damaged have the right cause
++            CraftEventFactory.blockDamage = blockDamager; // SPIGOT-6639: Re-set blockDamage so that other entities which are also getting damaged have the right cause // Paper
+ 
+             if (!event.isCancelled()) {
+                 event.getEntity().setLastDamageCause(event);
+@@ -1078,6 +1087,8 @@ public class CraftEventFactory {
+                 throw new IllegalStateException(String.format("Unhandled damage of %s by %s from %s", entity, damager.getHandle(), source.msgId));
+             }
+             EntityDamageEvent event = new EntityDamageByEntityEvent(damager, entity.getBukkitEntity(), cause, modifiers, modifierFunctions, source.isCritical()); // Paper - add critical damage API
++            */
```

Comment:
These lines were undeleted by purpose. I wasn't sure if I can safely delete these lines. If it is ok to delete this lines, I will.